client/my-sites/checkout/checkout-thank-you/marketplace/marketplace-thank-you.tsx

```diff
@@ -60,8 +62,8 @@ const MarketplaceThankYou = ( { productSlug }: { productSlug: string } ) => {
 		wpComPluginData ? wpComPluginData.software_slug || wpComPluginData.org_slug : productSlugs[ i ]
 	);
 
-	const pluginsOnSite: [] = useSelector( ( state ) =>
-		getPluginsOnSite( state, siteId, softwareSlugs )
+	const pluginsOnSite = useSelector( ( state ) =>
+		siteId ? getPluginsOnSite( state, siteId, softwareSlugs ) : []
```

Comment:
Yes, but it would be an array of undefined because `getPluginOnSite` returns undefined in that case. I think in this case most code would handle an empty array better than an array of undefined.

Reply:
In that case, should we just call `.filter( Boolean )` on the result of `.map()` to ensure we don't have any `undefined` items in the array? That would also result in an empty array if all of the items are undefined.

Reply:
What makes that method preferable to this one? The ternary operator results in less computation, although it's all negligible overall.

Reply:
It's about simplicity mostly. Which I prefer, unless you can demonstrate that it makes things slower 😉